tpp-app/src/services/OnboardingService.js

```diff
@@ -0,0 +1,102 @@
+
+const OnboardingService = {
+    /**
+     * Initializes user's initial period length and average period length. 
+     * @param periodLength number representing the period length
+     * @returns a promise resolving when the two set operations are complete 
+     */
+    PostInitialPeriodLength: async function(periodLength) {
+        try {
+            if(periodLength != null) {
+                periodLength = JSON.stringify(periodLength)
+                const initialRes = await AsyncStorage.setItem(KEYS.INITIAL_PERIOD_LENGTH, periodLength);
+                const averageRes = AsyncStorage.setItem(KEYS.AVERAGE_PERIOD_LENGTH, periodLength);
+
+                console.log('Initialized initialPeriodLength and averagePeriodLength as ' + periodLength);
+                return Promise.all([initialRes, averageRes]);
+            }
+        } catch (e) {
+            console.log('There was an error with POSTInitialPeriodLength');
+            console.log(e);
+        }
+    },
+    /**
+     * Initializes entries for the user's last period.  
+     * This only happens when the user has given inputs for both their 
+     * initial period length and the start date of their last period. 
+     * @param periodStart date representing the start date of their last period 
+     * @returns a promise resolving when the set operation is complete 
+     */
+    PostInitialPeriodStart: async function(periodStart) {
+        try {
+            const periodLength = await AsyncStorage.getItem('initialPeriodLength');
+            if(periodLength != null && periodStart != null) {
+                const periodStartTime = periodStart.getTime();
+                let yearsSet = new Set(); 
+                let dates = [];
+
+                // Use timestamps to populate the list of dates to avoid 
+                // messing up when dealing with last day of a month and/or year
+                for(let i = 0; i < periodLength; i++) {
+                    const currentDate = new Date(periodStartTime + (1000 * 60 * 60 * 24 * i));
+                    yearsSet.add(currentDate.getFullYear());
+                    dates.push(currentDate);
+                }
+                
+                let yearDicts = {};
+                yearsSet.forEach(year => {
+                    yearDicts[year] = initializeEmptyYear(year);
+                });
+
+                // Initialize the correct daily logs 
+                dates.forEach(date => {
+                    yearDicts[date.getFullYear()][date.getMonth()][date.getDate() - 1] = new Symptoms(FLOW_LEVEL.MEDIUM);
+                });
+
+                let allRes = [];
+                yearsSet.forEach(async (year) => {
+                    const res = await AsyncStorage.setItem(JSON.stringify(year), JSON.stringify(yearDicts[year]));
+                    allRes.push(res);
+                })
+                console.log("Initialized the first loggings");
+                return Promise.all(allRes);
```

Comment:
`Promise.all` is typically used when you have multiple promises to resolve for. Since we technically only have one here, I recommend changing your function to be
```javascript
const PostInitialPeriodStart = async (periodStart) => new Promise( (resolve, reject) => {
  try {
    ...
    resolve('yay it worked');
  } catch {
    ...
    reject();
  }
});
```
This will also handle my comment below about having a `reject` in the catch block.

Reply:
I'm not too familiar with Promises so thanks for all of this Emily! I've added rejects to all methods except here I used `Promise.all` to account for the possibility that I have to call `setItem()` more than once in case I need to initialize a 5-day-long period that started on December 30th of some year. Would the best practice then be to still use `Promise.all` or is there something else that I should use?